### DISCUSSION TITLE
Nginx configuration

### DISCUSSION THREAD

**Discussion by webspecialist** [vercel/next.js#22603] [Help]
How can I configure Nginx for nextJS?
Is there any documentation?

I tried to ask here:
[stack](https://stackoverflow.com/questions/66403877/next-js-change-directly-from-url-not-working)

**Comment by randrade-digital-fruit** [comment]
If your route is statics, yo have to define it in next.config.js file into a exportMapPath, the problem become when is dynamic, I had that problem and is impossible to me foun a solution.

**Reply by webspecialist** [reply]
I tried to serve with static out folder and pm2.

`
```
module.exports = {
  trailingSlash: true
}

module.exports = {
  sassOptions: {
    includePaths: [path.join(__dirname, 'styles')],
  },
}

module.exports = {
  async redirects() {
    return [
      {
        source: '/',
        destination: '/el',
        permanent: false,
      },
    ]
  },
}
```
`
It didn't listen to the next.config.js file for some reason.
The only way to work is with the custom server of node.js.

Thank you I will try this

**Comment by randrade-digital-fruit** [comment]
exportPathMap allows you to specify a mapping of request paths to page destinations, to be used during export. Paths defined in exportPathMap will also be available when using next dev.

Open next.config.js and add the following exportPathMap config:

```
module.exports = {
  exportPathMap: async function (
    defaultPathMap,
    { dev, dir, outDir, distDir, buildId }
  ) {
    return {
      '/': { page: '/' },
      '/about': { page: '/about' },
      '/p/hello-nextjs': { page: '/post', query: { title: 'hello-nextjs' } },
      '/p/learn-nextjs': { page: '/post', query: { title: 'learn-nextjs' } },
      '/p/deploy-nextjs': { page: '/post', query: { title: 'deploy-nextjs' } },
    }
  },
}
```

[HERE](https://nextjs.org/docs/api-reference/next.config.js/exportPathMap) is the documentation

The 'problem' here, at least in my case, is that nginx doesn't support dymanics routes, or if it does I have to congifure settings and I dont know how to do it, if your app is static it's fine, but with a dynamic appp is better found other ways like vercel or some server that support a node.js and use npm start.

**Reply by webspecialist** [reply]
Thank you very much. I understand what the problem was in my case.
My site is completely based on dynamic routes. So I would use node.js
